### ISSUE TITLE
Remove dependency requirements for OSX payloads

### ISSUE THREAD

**Issue by Viss** [n1nj4sec/pupy#81]
when trying to shell a mac, one must run pp.py, which has a bunch of dependencies, and presumes that the shell obtained on a host likely won't have homebrew installed, or may already have a dev environment setup or any other number of circumstances, so payloads for osx should be able to run without the requirement for any dependencies. 

it would be amazing if pupy was able to create mach-o binaries, or python code that would function independently without extra requirements from the victim - perhaps similar to how meterpreter functions where it has a 'staged' payload and the binary component can be downloaded. 

short version: needing to install deps on a victim host makes things extremely difficult.

**Comment by mandreko** [comment]
This is exactly what pyinstaller is for. https://github.com/n1nj4sec/pupy/wiki/Generate-payloads#on-osx
